BE/src/main/java/com/example/User/UserController.java

```diff
@@ -160,19 +133,35 @@ public ResponseEntity<?> register(@RequestBody RegisterRequest registerRequest)
     }
 
     @Operation(summary = "Get all users")
-    @ApiResponses(
-            value = {
-                @ApiResponse(
-                        responseCode = "200",
-                        description = "Users retrieved successfully",
-                        content = {
-                            @Content(
-                                    mediaType = "application/json",
-                                    array = @ArraySchema(schema = @Schema(implementation = UserDTO.class)))
-                        })
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Users retrieved successfully", content = {
+                    @Content(mediaType = "application/json", array = @ArraySchema(schema = @Schema(implementation = UserDTO.class)))
             })
+    })
     @GetMapping
     public ResponseEntity<List<UserDTO>> getAll() {
         return ResponseEntity.ok(userService.getAll());
     }
+
+    @Operation(summary = "Get the energy level of a user.")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Energy retrieved successfully.", content = {
+                    @Content(mediaType = "application/json", schema = @Schema(implementation = Integer.class))
+            })
+    })
+    @GetMapping("/{id}/energy")
+    public ResponseEntity<Integer> getEnergy(@PathVariable long id) {
+        return ResponseEntity.ok(userService.getById(id).getEnergy());
+    }
+
+    @Operation(summary = "Set the energy level of a user.")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Energy set successfully.")
+    })
+    @PatchMapping("/{id}/energy")
+    public ResponseEntity<Integer> setEnergy(@PathVariable long id, @RequestBody EnergyUpdateRequest updateRequest) {
+        userService.setEnergy(id, updateRequest.getEnergy());
+        return ResponseEntity.ok().build();
+    }
+
```

Comment:
I'd extend this to support patches of other values like so:
```suggestion
    @PatchMapping("/{id}")
    public ResponseEntity<UserDTO> updateUserData(@PathVariable Long id, @RequestBody UserDTO updateRequest) {
        return ResponseEntity.ok(userService.update(updateRequest)).build();
    }
```
This is a more RESTful solution as PATCH should be general enough to handle any kind of field changes. It also keeps the controller ergonomic.

Reply:
One could argue "why do we have that PUT handler then?" I think we should use that for handling larger user-related changes like those relating to credentials. It's not entirely RESTful but if we want it to be we could also change the PUT handler to be a PATCH handler instead for consistency.